### PR TITLE
wasi: support monotonic clock on clock_time_get.

### PR DESCRIPTION
**Pull request by mathetake**
Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

resolves https://github.com/proxy-wasm/proxy-wasm-rust-sdk/issues/103

### CODE REVIEW

File: src/exports.cc
Comment (PiotrSikora) [inline src/exports.cc:849]: That's not the best way to implement this. Instead, you should return `Envoy::RealTimeSource::monotonicTime()`,  which requires plumbing this via `Context`.

**Review by PiotrSikora — CHANGES_REQUESTED** [review]

File: src/exports.cc
Comment (mathetake) [reply src/exports.cc]: thanks!

File: test/exports_test.cc
Comment (PiotrSikora) [inline test/exports_test.cc:51]: Could you do this refactoring as part of a separate PR? It doesn't look necessary here and it makes reviewing this specific change much harder when it's intermixed with unrelated changes. Thanks!

File: include/proxy-wasm/context.h
Comment (PiotrSikora) [inline include/proxy-wasm/context.h:260]: I guess this is fine for now, since it follows `getCurrentTimeNanoseconds()`, but we should remove those default implementations from headers, to avoid accidentally leaking them into production code... and then we can have a dummy `Context` implementation used for tests in this repo.

File: test/exports_test.cc
Comment (mathetake) [reply test/exports_test.cc:51]: yeah right. Will revert the test refactoring change. Thanks for your patience.

File: include/proxy-wasm/context.h
Comment (mathetake) [reply include/proxy-wasm/context.h:260]: OK, will do the cleanup in https://github.com/proxy-wasm/proxy-wasm-cpp-host/issues/158 later, so I'm leaving this as-is for now.

**Review by PiotrSikora — APPROVED** [review]
LGTM, thanks! Please merge only after the relevant Envoy PR is approved and passes tests.

**Pull request merged**
